packages/itwin/grouping-mapping-widget/src/widget/components/GroupPropertyAction.tsx

```diff
@@ -204,55 +310,135 @@ const extractProperties = (
-                type: "string",
-              });
-
-              classToPropertiesMapping.get(className)?.push({
-                name: "TypeDefinition.Model.ModeledElement.CodeValue",
-                label: "Model CodeValue (TypeDefinition)",
-                type: "string",
-              });
-
+              ecPropertyMetaDetaList.push(...extractNested(["TypeDefinition"], nestedContentField.nestedFields));
             }
             break;
           }
           default: {
             // Some elements don't have a path to primary class or relationship meaning..
             // Most likely a simple struct property
             if (!nestedContentField.pathToPrimaryClass) {
-              const columnName = (property as PropertiesField).properties[0]
-                .property.name;
-              const className = (property as PropertiesField).properties[0]
-                .property.classInfo.name;
-              extractStructProperties(
-                navigation
-                  ? `${navigation.navigationName}.${columnName}`
-                  : columnName,
-                navigation ? navigation.rootClassName : className,
-                classToPropertiesMapping,
-                property.type.members
-              );
-
+              extractStruct(property, ecPropertyMetaDetaList);
             }
           }
         }
       }
     }
   }
+  return ecPropertyMetaDetaList;
+};
+
+const convertToECProperties = (property: PropertyMetaData): Array<ECProperty> => {
+  const ecProperty: ECProperty = {
+    ecSchemaName: property.schema,
+    ecClassName: property.className,
+    ecPropertyType: property.type,
+    ecPropertyName: "",
+  };
+  switch (property.primitiveNavigationClass) {
+    // Hardcode Models navigation path and label behavior
+    case "BisCore:ModelContainsElements": {
+      return [
+        {
+          ...ecProperty,
+          ecPropertyName: [
+            ...property.propertyTraversal,
+            "ModeledElement",
+            "UserLabel",
+          ].join("."),
+        },
+        {
+          ...ecProperty,
+          ecPropertyName: [
+            ...property.propertyTraversal,
+            "ModeledElement",
+            "CodeValue",
+          ].join("."),
+        },
+      ];
+    }
+    // Hardcode Category navigation path and label behavior
+    case "BisCore:GeometricElement3dIsInCategory":
+      return [
+        {
+          ...ecProperty,
+          ecPropertyName: [...property.propertyTraversal, "UserLabel"].join("."),
+        },
+        {
+          ...ecProperty,
+          ecPropertyName: [...property.propertyTraversal, "CodeValue"].join("."),
+        },
+      ];
+    // Hardcode Material path and label behavior
+    case "BisCore:PhysicalElementIsOfPhysicalMaterial": {
+      return [
+        {
+          ...ecProperty,
+          ecPropertyName: [...property.propertyTraversal.slice(0, -1), "Material", "UserLabel"].join("."),
```

Comment:
Why the `slice` for Material?

Reply:
The property name that Presentation returns is PhysicalMaterial. I slice it out and replace it with Material because that is the actual column name.

Reply:
This looks like a very weird and unnecessary edge case. See the BisCore schema: https://github.com/iTwin/bis-schemas/blob/master/Domains/0-Core/BisCore.ecschema.xml

PhysicalMaterial everywhere. How can it be the "actual column name"?

Reply:
Good catch. I'm not sure how I arrived to "Material". Must have confused another schema when I looked into this a while back.

Reply:
done